flutter/lib/resources/result_manager.dart

```diff
@@ -9,7 +50,71 @@ class ResultManager {
     _jsonResultPath = '$applicationDirectory/$_jsonResultFileName';
   }
 
-  Future<void> write(List<Map<String, dynamic>> content) async {
+  // returns brief results in json
+  Future<String> record(List<RunResult?> results) async {
+    final resultContent = <Map<String, dynamic>>[];
+    final briefResultContent = <Map<String, dynamic>>[];
+
+    for (final result in results) {
+      if (result == null) continue;
+      final score = result.mode == BenchmarkMode.accuracy ? null : result.score;
+
+      final benchmarkResult = {
+        'benchmark_id': result.id,
+        'configuration': {
+          'runtime': '',
+        },
+        'score': score != null ? score.toString() : 'N/A',
+        'accuracy': BenchmarkMode.performance_lite == result.mode
+            ? 'N/A'
+            : result.accuracy,
+        // strings are used here to match android app behavior
+        'min_duration': result.minDuration.toString(),
+        'duration': result.durationMs.toString(),
+        'min_samples': result.minSamples.toString(),
+        'num_samples': result.numSamples.toString(),
+        'shards_num': result.threadsNumber,
+        'batch_size': result.batchSize,
+        'mode': result.mode.toString(),
+        'datetime': DateTime.now().toIso8601String(),
+      };
+      var briefResult = _BriefResult(result.id, score, result.accuracy,
```

Comment:
Since there is a `_BriefResult`, can you also create a `_BenchmarkResult`.

Reply:
What would be the benefit?
I created `_BriefResult` class because we parse it. It just doesn't look good when we define 2 equivalent string constants to write and then to parse json in different places in code, and use quite different logic for reading and writing, which can result in reading not what we thought we wrote. Looked kinda brittle.
We don't parse full results json anywhere, we just use for sharing. Creating a class for it would increase number of line but wouldn't really improve anything in my opinion.

Maybe we should just move json serialization of full results to a separate method?
Or we could just delete `_BriefResult` class and save results in the same format that we use for sharing.

Reply:
Both `BriefResult` and `BenchmarkResult` are data models. I think it's best practice to define data model as a `class` (or `struct` depend on language). Methods for parsing/storing can be placed inside `ResultManager` or we can make both `BriefResult` and `BenchmarkResult` conform to a JSON serialization / deserialization protocol and `ResultManager` only handles the reading from disk or storing to disk.

Reply:
done